integration-libs/opf/public_api.ts

```diff
@@ -0,0 +1,8 @@
+/*
+ * SPDX-FileCopyrightText: 2024 SAP Spartacus team <spartacus-team@sap.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+export * from '@spartacus/opf/base';
+export * from '@spartacus/opf/checkout';
```

Comment:
[MAJOR]

After reviewing the whole epic branch, I understood that we deal with 5 distinct areas:
- CTA 
- Checkout (with Payment, Payment Verification and Placing Order) 
- Quick Buy (aka "buy without a regular checkout")
- Payment Gateway Integrations like ApplePay,GooglePay,PayPal,... [which can be used by Quick Buy _and potentially by Checkout_]
- Utils (Global Functions, Scripts Loader) [which can used by CTA and Checkout]

I attach here a diagram of the proposed architecture of dependencies.

![image](https://github.com/SAP/spartacus/assets/4001059/171e827b-602d-4c73-8f7b-ffa53ac58d46)


---

Unfortunately, currently in the epic branch we have cross-dependencies in between distinct areas, which I have problem to understand a justification for:
- QuickBuy [depends on](https://github.com/SAP/spartacus/blob/bf4a1c2cdb552b16adbd1b34a12789d32b9b0175/integration-libs/opf/base/components/opf-quick-buy/opf-quick-buy.service.ts#L27-L28) Payments (but IMHO Payments should be scoped only to the Checkout)
- ApplePay [depends on](https://github.com/SAP/spartacus/blob/bf4a1c2cdb552b16adbd1b34a12789d32b9b0175/integration-libs/opf/base/components/opf-quick-buy/apple-pay/apple-pay.service.ts#L46-L55) Payments and QuickBuy . [Analogically for GooglePay](https://github.com/SAP/spartacus/blob/bf4a1c2cdb552b16adbd1b34a12789d32b9b0175/integration-libs/opf/base/components/opf-quick-buy/google-pay/google-pay.service.ts#L35-L41). (but IMHO Payments should be scoped only to the Checkout; and ApplePay should be used by QuickBuy, but not the other way around)

Reply:
[OUTDATED] Done during refactoring.